docs/cookbook/creating-custom-schema-fields.md

```diff
@@ -0,0 +1,999 @@
+# Building custom schema fields
+
+## Introduction
+Apostrophe comes with a large number of [schema field types](/reference/field-types/). Many times when we need a custom field type, it is simply a combination of existing field types working together for a new outcome. Luckily, Apostrophe was designed to allow you to reuse existing field types to create new custom fields.
+
+In this recipe, we are going to build two different custom schema fields. The first will be a simple set of `string` and `integer` fields to collect grade information. The second will be a gradient background creator with `integer` and `color` fields, within an `array`. Overall, the steps to building both will be almost identical. However, we are going to build the grade field as part of a piece and the gradient as a stand-alone module. This will demonstrate the changes you need to make when building your custom schema field in a module that doesn't extend another core module and changes that need to be made to `object` and `array` schema input fields when you use them this way.
+
+## Structure of a schema field
+
+A schema input field has two parts: a server-side part and a browser-side part.
+
+The server-side code handles registering the schema field type, plus sanitizing and storing the data in the database. Most of this code will be the same for each custom field. We can take advantage of the same code that the Apostrophe core uses for sanitization and storage.
+
+The browser-side code provides the admin UI and is where we will take advantage of the built-in fields to lighten the programming load. Apostrophe has two Vue components and two mixin files to assist with schema creation. The first component is the `AposInputWrapper`. This component helps with field validation and appearance. The other is the `AposSchema` component. This Vue component parses through a supplied schema of fields and implements the v-model that allows two-way data binding between a value in our template and a value in our data properties. Apostrophe provides the `AposInputMixin` mixin to facilitate passing props, display modifications, and event listeners. The `AposInputChoicesMixin` file assists in creating dynamic and static choices if you are using the `checkboxes`, `radio`, or `select` fields in your custom schema.
```

Comment:
Last sentence is incorrect. `AposInputChoicesMixin` is useful if you are implementing a new field type *similar to* our checkboxes, radio or select fields, e.g. one that presents a choice of one or more options, and you want support for loading those options dynamically via an API call.

Reply:
Eliminated.